pages/index.vue

```diff
@@ -1609,46 +1487,19 @@ export default {
     },
     setRouteQueryState() {
       const flat = (key) => (this[key] !== "" ? `${key}=${this[key]}&` : "")
-      const deep = (key) => {
-        const haveItems = [...this[key]].length
-        if (haveItems && this[key]["value"] !== "") {
-          // Exclude files fro  query params
-          const filesRemoved = this[key].filter((item) => !(item?.value?.[0] instanceof File))
-          return `${key}=${JSON.stringify(filesRemoved)}&`
-        }
-        return ""
-      }
-      let flats = [
-        "method",
-        "url",
-        "path",
-        !this.URL_EXCLUDES.auth ? "auth" : null,
-        !this.URL_EXCLUDES.httpUser ? "httpUser" : null,
-        !this.URL_EXCLUDES.httpPassword ? "httpPassword" : null,
-        !this.URL_EXCLUDES.bearerToken ? "bearerToken" : null,
```

Comment:
We need to introduce `URL_EXCLUDES` list so that all request object attributes don't have to be set as a URL query in hoppscotch.io?_______ URL.

```
https://deploy-preview-1567--hoppscotch.netlify.app/?url=https://httpbin.org&path=/get&request=%7B%22name%22:%22Untitled%20request%22,%22method%22:%22GET%22,%22uri%22:%22https://httpbin.org/get%22,%22url%22:%22https://httpbin.org%22,%22path%22:%22/get%22,%22auth%22:%22None%22,%22httpUser%22:%22%22,%22httpPassword%22:%22%22,%22passwordFieldType%22:%22password%22,%22bearerToken%22:%22%22,%22headers%22:%5B%5D,%22params%22:%5B%5D,%22bodyParams%22:%5B%5D,%22rawParams%22:%22%22,%22rawInput%22:true,%22requestType%22:%22%22,%22contentType%22:%22%22,%22status%22:200,%22date%22:%2228/03/2021%22,%22time%22:%2222:47:52%22,%22updatedOn%22:%222021-03-28T17:17:52.391Z%22,%22usesPreScripts%22:true,%22preRequestScript%22:%22//%20pw.env.set(%27variable%27,%20%27value%27);%22,%22duration%22:290,%22star%22:false,%22testScript%22:%22//%20pw.expect(%27variable%27).toBe(%27value%27);%22,%22usesPostScripts%22:true
```

So this is a typical URL after refactor - on a second thought - instead of `URL_EXCLUDES` list, I guess it's far better to introduce a `URL_INCLUDES` list with only the attributes we want to have in URL parameters.

Request attributes like `name`, `star`, `duration` etc.. are absolute to be in URL.

Reply:
One edge case is that: in authentication section, we allow users to include / exclude auth credentials in URL or not. Default is exclude. And this is basically the very reason why there's a `URL_EXCLUDES` list in the first place.

<img width="658" alt="Screenshot 2021-03-28 at 11 42 55 PM" src="https://user-images.githubusercontent.com/10395817/112762814-a598b380-8fb6-11eb-94c8-d29482dda318.png">

Reply:
@liyasthomas is this feature only for authentication or for all attributes?

Reply:
Not just Authentication attributes but also attributes such as duration, size etc. which are not essential to be a part of URL query.